src/main/java/org/mifos/connector/crm/routes/BillPayRouteBuilder.java

```diff
@@ -28,8 +34,21 @@ public void configure() {
         from("direct:bill-payments").routeId("bill-payments").log("Received request for bill payments").unmarshal()
                 .json(JsonLibrary.Jackson, BillPaymentsReqDTO.class).setHeader(Exchange.HTTP_RESPONSE_CODE, constant(200))
                 .process(exchange -> {
+                    BillPaymentsResponseDTO response;
                     logger.debug("Bill Payments Request: " + exchange.getIn().getBody(BillPaymentsReqDTO.class));
-                    BillPaymentsResponseDTO response = setResponseBody(exchange.getIn().getBody(BillPaymentsReqDTO.class));
+                    if(exchange.getIn().getBody(BillPaymentsReqDTO.class).getBillId().equals(billAlreadyPaidId)){
+                        response =
+                                setResponseBodyForBillPaid(exchange.getIn().getBody(BillPaymentsReqDTO.class));
+                    }
+                    else if(exchange.getIn().getBody(BillPaymentsReqDTO.class).getBillId().equals(billPayTimeoutId)){
+                        Thread.sleep(15000);
```

Comment:
is this Thread.sleep added to mock the "time out" scenario?

Reply:
yes